src/components/Reown/modals/BaseNanoContractModal.js

```diff
@@ -0,0 +1,374 @@
+                      {argType && <small className="text-muted d-block">{argType}</small>}
+                    </td>
+                    <td className="border-top-0 text-monospace" style={{ wordBreak: 'break-all' }}>
+                      {nanoUtils.formatNCArgValue(arg, argType, decimalPlaces)}
+                    </td>
+                  </tr>
+                );
+              })}
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+/**
+ * Component for Caller Address Section
+ */
+const CallerAddressSection = ({ selectedAddress, onSelectAddress, nanoContracts, nanoContract, dispatch }) => {
+  const nanoContractsRegisterMetadata = useSelector((state) => state.nanoContractsRegisterMetadata);
+  const isRegistering = nanoContractsRegisterMetadata?.status === 'loading' && nanoContractsRegisterMetadata?.ncId === nanoContract.ncId;
+  const registrationError = nanoContractsRegisterMetadata?.status === 'error' && nanoContractsRegisterMetadata?.ncId === nanoContract.ncId;
+
+  return (
+    <div className="mb-3">
+      <strong>{t`Caller`}</strong>
+      <div className="d-flex align-items-center">
+        <div className="text-monospace flex-grow-1">{selectedAddress || '-'}</div>
+        <button
+          className="btn btn-link btn-sm p-0 ml-2"
+          onClick={onSelectAddress}
+          title={t`Select Address`}
+        >
+          <i className="fa fa-pencil" style={{ fontSize: '1.2rem' }}></i>
+        </button>
+      </div>
+
+      {nanoContract.ncId && !nanoContracts[nanoContract.ncId] && (
+        <div className="alert alert-info mt-3 mb-0">
+          <i className="fa fa-info-circle mr-2"></i>
+          {t`This nano contract is not registered in your wallet. Would you like to register it?`}
+          <div className="mt-2">
+            <button
+              className="btn btn-sm btn-outline-primary"
+              onClick={() => dispatch({
+                type: types.NANOCONTRACT_REGISTER_REQUEST,
+                payload: {
+                  ncId: nanoContract.ncId,
+                  address: selectedAddress
+                }
+              })}
+              disabled={!selectedAddress || isRegistering}
+            >
+              {isRegistering ? t`Registering...` : t`Register Nano Contract`}
+            </button>
+          </div>
+          {registrationError && (
+            <div className="text-danger mt-2">
+              <small>{nanoContractsRegisterMetadata.error}</small>
+            </div>
+          )}
+        </div>
+      )}
+    </div>
+  );
+};
+
+/**
+ * Component for dApp Info
+ */
+const DAppInfo = ({ dapp }) => (
+  <div className="d-flex align-items-center mb-4">
+    {dapp?.icon && (
+      <img
+        src={dapp.icon}
+        alt="dApp icon"
+        className="mr-3"
+        style={{ width: 48, height: 48 }}
+      />
+    )}
+    <div>
+      <h6 className="mb-1">{dapp?.proposer || "Unknown"}</h6>
```

Comment:
question(non-blocking): In the case of not having a proposer, will the user see enough context in the screen to understand what this `Unknown` is about?

Reply:
Makes sense. Maybe add more context "Unknown dapp proposer"

And maybe this component could be in a separate file, it seems it could be reused by other components as well.

And it's missing the translation tag.

Reply:
Refactored it, we had duplicated code in all modals. Thanks!

Also changed to "Unknown Dapp Proposer" with ttags